### ISSUE TITLE
L2X for hierarchical LSTM returns always first sentence as selection

### ISSUE THREAD

**Issue by ikonstas-ds** [Jianbo-Lab/L2X#4]
Hello and congratulations for your great work.
I tried to implement the L2X explanations with hierarchical LSTM on a fake news dataset, as shown in the imdb_sent example. However, the explainer always selects the first sentence as an explanation. Do you have any idea why is this happening? 
I would also like to ask you (sorry if it is a stupid question), as the method is model agnostic, why do you use different neural network architectures for each imdb example?

**Comment by Jianbo-Lab** [comment]
1. the explainer always selects the first sentence as an explanation. Do you have any idea why is this happening?
I guess this is because the explainer is not well trained. In particular, training is sensitive to the hyper-parameter for Gumbel Softmax distribution. 

2. As the method is model agnostic, why do you use different neural network architectures for each imdb example?
The lower bound (between Eq. 3 and 4 here: https://arxiv.org/pdf/1802.07814.pdf) can yield a better approximation if the neural network is chosen properly. Without assuming such knowledge, the method might still work.
